conda_build/build.py

```diff
@@ -1915,10 +1924,26 @@ def bundle_conda(output, metadata: MetaData, env, stats, **kw):
     if files:
         # Files is specified by the output
         # we exclude the list of files that we want to keep, so post-process picks them up as "new"
-        keep_files = {
-            os.path.normpath(pth)
-            for pth in utils.expand_globs(files, metadata.config.host_prefix)
-        }
+        if isinstance(files, dict):
+            # When file matching with include/exclude lists, only
+            # new_prefix_files are considered. Files in the PREFIX from other
+            # recipes (dependencies) are ignored
+            include = files.get("include", [])
+            exclude = files.get("exclude", [])
+            exclude_files = {
+                os.path.normpath(pth)
+                for pth in utils.expand_globs(exclude, metadata.config.host_prefix)
+            }
+            keep_files = {
+                os.path.normpath(pth)
+                for pth in utils.expand_globs(include, metadata.config.host_prefix)
+            }
+            keep_files = new_prefix_files.intersection(keep_files) - exclude_files
```

Comment:
Just to make sure, the precedence of operators here is:

* From all new files added to PREFIX
* Keep the ones seen in `include`
* And then remove those in `exclude`

My set theory is rusty so, again, just to double check, is it the same if we diff `keep_files` and `exclude_files` first, and then intersect the result with `new_prefix_files`?

Reply:
The set difference would be the intersection with the negated set, so
(A ∩ B ) \ C = (A ∩ B ) ∩ C' = A ∩ B ∩ C' = A ∩ (B ∩ C') = A ∩ (B \ C)
. Or to put it another words: We end up with intersection the same sets (set intersection is associative).

Reply:
Yes; it's the same. I could have also implemented it as you have described:

```suggestion
            keep_files = new_prefix_files.intersection(keep_files - exclude_files)
```

Reply:
Thanks for the explanations! Marking this as resolved then.